modules/Bio/EnsEMBL/DBSQL/SliceAdaptor.pm

```diff
@@ -2727,6 +2727,84 @@ sub _build_circular_slice_cache {
   $sth->finish();
 } ## end _build_circular_slice_cache
 
+=head2 create_chromosome_alias
+
+  Args       : none
+  Example    : $self->create_chromosome_alias();
+  Description: create chromosome alias in coordinate system with karyotype attributes
+  Returntype : Bio::EnsEMBL::CoordSystem, or none 
+  Exceptions : None
+  Caller     : general
+  Status     : in testing
+
+=cut
+
+sub _create_chromosome_alias {
+  my $self = shift;
+  my $csa  = $self->db->get_CoordSystemAdaptor();
+
+  # to store reformatted db query results
+  my %karyotype_seq_regions;
+  
+  my $karyotype_rank_string = "karyotype_rank";
+
+  # to store coord_system_id to alias 
+  my $coord_system_id;
+
+  # check whether seq region cache exists, otherwise run database query
+  if ( $self->{'karyotype_cache'} ) {
+    %karyotype_seq_regions = $self->{'karyotype_cache'};
+  } else {
+    # cannot find a coordssystem called chromosome
+    # look through attribs to find seq_regions with karyotype
+
+    my $sth =
+      $self->prepare( "SELECT sr.seq_region_id, sr.name, sr.coord_system_id, sr.length, a.code "
+                    . "FROM seq_region sr, seq_region_attrib sra, attrib_type a "
+                    . "WHERE sr.seq_region_id = sra.seq_region_id "
+                    . "AND a.attrib_type_id = sra.attrib_type_id "
+                    . "AND a.code = ?" );
+
+    $sth->bind_param( 1, $karyotype_rank_string );
+    $sth->execute();
+
+    # fetch SQL results and identify coord_system_id that
+    # has karyotype attribs
+
+    while ( my $hashref = $sth->fetchrow_hashref() ) {
+      my $seq_region_id   = $hashref->{ seq_region_id };
+      my $seq_region_name = $hashref->{ name };
+      $coord_system_id    = $hashref->{ coord_system_id };
```

Comment:
I would check to make sure that you don't get more than one coordinate system here. In all the examples we've seen so far, there is only one coord_system that has karyotypes (only one coord_system at all, for that matter). However, I'm not sure this is guaranteed.

Reply:
Added check in with commit https://github.com/Ensembl/ensembl/pull/532/commits/c4259bedaab2992ae177ba28d4959a70b5b9ba0e